src/Datasource/QueryInterface.php

```diff
@@ -159,7 +159,7 @@ public function count(): int;
      * $query->limit($query->newExpr()->add(['1 + 1'])); // LIMIT (1 + 1)
      * ```
      *
-     * @param int $num number of records to be returned
+     * @param int|mixed $num number of records to be returned
```

Comment:
The mixed here sounds bad - as an addition it makes it pretty much worthless to expect and input type.

Reply:
Datasource doesn't contain a way to describe a QueryExpression which is what the database implementation can accept in addition to an int. Ither implementations may accept othe 'expression' style objects as well.